server/src/api/modules/name-changes/services/FindGroupNameChangesService.ts

```diff
@@ -24,11 +24,18 @@ async function findGroupNameChanges(
     return [];
   }
 
-  // Fetch all achievements for these player IDs
+  // Fetch all name changes for these player IDs
   const nameChanges = await prisma.nameChange.findMany({
     where: {
       playerId: { in: playerIds },
-      status: NameChangeStatus.APPROVED
+      status: NameChangeStatus.APPROVED,
+      player: {
```

Comment:
What about the general `/names` endpoint?

Reply:
Added filtering to that endpoint too now.